docs/proposals/cron-wf-improvement-proposal.md

```diff
@@ -0,0 +1,102 @@
+# Proposal for Cron Workflows improvements
+
+## Introduction
+
+Currently, CronWorkflows are a great resource if we want to run recurring tasks to infinity. However, it is missing the ability to customize it, for example define how many times a workflow should run or how to handle multiple failures. I believe argo workflows would benefit of having more configuration options for cron workflows, to allow to change its behavior based on the result of its child’s success or failures. Below I present my thoughts on how we could improve them, but also some questions and concerns on how to properly do it.
+
+## Proposal
+
+This proposal discusses the viability of adding 2 more fields into the cron workflow configuration:
+
+```
+RunStrategy:
+ maxSuccess:
+ maxFailures:
+```
+
+`maxSuccess` - defines how many child workflows must have success before suspending the workflow schedule
+
+`maxFailures` - defines how many child workflows must fail before suspending the workflow scheduling. This may contain `Failed` workflows, `Errored` workflows or spec errors.
```

Comment:
very important to document this clearly in the docs when the feature is released.